python/doc/examples/reliability_sensitivity/design_of_experiments/plot_smolyak_indices.py

```diff
@@ -0,0 +1,102 @@
+"""
+Plot Smolyak multi-indices
+==========================
+"""
+# %%
+# The goal of this example is to plot the multi-indices used in Smolyak's
+# quadrature.
+#
+# For a given dimension :math:`d_x \in \mathbb{N}` and a given level
+# :math:`\ell \in \mathbb{N}`
+# Smolyak's quadrature is the combination of tensorized univariate quadratures.
+# These quadrature are defined by the set of multi-indices:
+#
+# .. math::
+#
+#     \mathcal{S}_{\ell, d_x} = \left\{\|\boldsymbol{k}\|_1 \leq \ell
+#                                      + d_x - 1\right\}
+#
+# where :math:`\|\boldsymbol{k}\|_1 = k_1 + ... + k_{d_x}` is the 1-norm of the
+# multi-index :math:`\boldsymbol{k} \in \mathbb{N}^{d_x}`.
+#
+# The goal of this script is to plot the multi-indices involved in Smolyak's
+# quadrature for different values of the level :math:`\ell` in
+# dimension :math:`d_x = 2`.
+
+# %%
+import openturns as ot
+import openturns.experimental as otexp
+import openturns.viewer as otv
+from matplotlib import pylab as plt
+
+# %%
+# In the first example, we print the indices involved in
+# Smolyak-Legendre quadrature of level 3.
+# The multi-indices are computed using the
+# :meth:`~openturns.SmolyakExperiment.computeCombination` method.
+# Actually, the multi-indices do not actually depend on the
+# underlying univariate quadratures, but this is required for
+# the :class:`~openturns.SmolyakExperiment` class.
+
+uniform = ot.GaussProductExperiment(ot.Uniform(0.0, 1.0))
+collection = [uniform] * 2
+level = 3
+print("level = ", level)
+experiment = otexp.SmolyakExperiment(collection, level)
+indices = experiment.computeCombination()
+print(indices)
+
+# %%
+# We see that the multi-indices have a sum which is equal to either 3 or 4.
+# In other words, these multi-indices belong to two different layers of
+# constant 1-norms.
+#
+# In order to see how this evolves depending on the level of the quadrature,
+# the following function creates a 2D plot of the set of multi-indices.
+
+
+def drawSmolyakIndices(level):
+    # Plot Smolyak indices of given level in 2 dimensions
+    uniform = ot.GaussProductExperiment(ot.Uniform(0.0, 1.0))
```

Comment:
uniform is a strange name for a GaussProductExperiment. BTW we don't care about the actual distribution. Why not simply put
```
collection = [ot.GaussProductExperiment()]*2
```

Reply:
Is this comment outdated because you took my remark into account? I can't see it in the code.